build_tools/travis/before_install.sh

```diff
@@ -28,6 +28,10 @@ if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
         brew install vtk
     fi
 
+    if [[ "$USE_OMP" == 1 ]]; then
+        brew install llvm
```

Comment:
you're still using xcode clang and not the one from brew. See

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang

in travis log.

you need to set flags for C and not only CPP files.